testutils/test_server.go

```diff
@@ -306,6 +309,11 @@ func (ts *TestServer) AssertRelayStats(expected *relaytest.MockStats) {
 	ts.relayStats.AssertEqual(ts, expected)
 }
 
+// ChannelOpts returns a copy of the server's configured ChannelOpts.
+func (ts *TestServer) ChannelOpts() *ChannelOpts {
```

Comment:
This is to support changes like [this one in #871](https://github.com/uber/tchannel-go/pull/871/files#diff-d29fd5cb46b394f16c180f4020e4022de2e7a73267e9558ddcf2679be6c7bcb4R178), where the correct frame pool was not being used; previously, one end (the de-pooling end, e.g. `pool.Get`) was using a checked frame pool, and the other end (the pooling end, e.g. `pool.Put`) was using a sync pool.

I originally kept it with these changes since it was just a basic "test server API change" that felt like it fit better with other misc test changes instead of polluting #871 with changes other than "turn on framepool checks". I can pull it into that PR instead, though.